File: A2209I/DMA/UserHub/UserHub/DTOs/Requests/User/LoginUserRequest.cs
using System.ComponentModel.DataAnnotations;

namespace UserHub.DTOs.Requests.User
{
    public class LoginUserRequest
    {
        [Required(ErrorMessage = "Email is required")]
        [EmailAddress(ErrorMessage = "A valid email is required")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Password is required")]
        [StringLength(100, ErrorMessage = "Password must be at least 6 characters long.", MinimumLength = 6)]
        public string Password { get; set; }

    }
}

File: A2209GDoiCan/IASF/demo/src/main/java/com/aptech/demo/dtos/requests/LoginUserRequest.java
package com.aptech.demo.dtos.requests;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.*;


@NoArgsConstructor
@AllArgsConstructor
@Builder //custom constructor

public class LoginUserRequest {

    @NotBlank(message = "Email is required.")
    @Email(message = "Invalid email format.")
    private String email;

    @NotBlank(message = "Password is required.")
    @Size(min = 4, message = "Password must be at least 4 characters long.")
    private String password;


    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}
